conpot/tests/test_mac_addr.py

```diff
@@ -0,0 +1,44 @@
+# Copyright (C) 2015  Adarsh Dinesh <adarshdinesh@gmail.com>
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+
+import unittest
+import conpot.utils.mac_addr
+import subprocess
+
+
+class TestMacAddrUtil(unittest.TestCase):
+
+    def setUp(self):
+        pass
+
+    def tearDown(self):
+        pass
+
+    def test_mac(self):
+        testmac = "00:de:ad:be:ef:00"
+        s = subprocess.Popen(["ifconfig"], stdout=subprocess.PIPE)
+        data = s.stdout.readlines()
+        for line in data:
+            if "Ethernet" in line:
+                break
+        if isinstance(line, basestring) and line:
```

Comment:
Yah, Got it.. '\n' was the problem. I checked if the string is empty but the last run of the loop will make the string '\n' which will go undetected. Adding split() to avoid. Let me check if there is a better way to get the interface names.